lineapy/graph_reader/artifact_collection.py

```diff
@@ -197,10 +206,8 @@ def _extract_session_module(
         )
 
         # Generate session function name
-        for coll in session_artifacts.artifact_nodecollections:
-            if coll.collection_type == NodeCollectionType.ARTIFACT:
-                first_art_name = coll.safename
-                break
+        first_art_name = session_artifacts._get_first_artifact_name()
+        assert isinstance(first_art_name, str)
```

Comment:
`assert first_art_name is not None` is better because you do not couple it to the return type of `_get_first_artifact_name`. Although Ideally some exception should be raised here.

Reply:
Thanks for the suggestion! Updates were made accordingly.

Reply:
@mingjerli So far, this is the only change in `artifact_collection.py` since your latest rebased version. As you can see, it is a very minor change, so there should not be much trouble rebasing again.